src/material/schematics/ng-generate/mdc-migration/rules/components/slider/slider-template.ts

```diff
@@ -25,4 +47,72 @@ export class SliderTemplateMigrator extends TemplateMigrator {
     });
     return updates;
   }
+
+  /** Returns all of the property, attribute, event, or two-way bindings on the given node. */
+  private _getBindings(node: compiler.TmplAstElement): Binding[] {
+    const allInputs = this._getInputs(node);
+    const allOutputs = this._getOutputs(node);
+    const attributes = this._getAttributes(node);
+    const twoWayBindings = this._getTwoWayBindings(allInputs, allOutputs);
+
+    // Remove the inputs & outputs that are two-way bindings.
+
+    const inputs = allInputs.filter(
+      input => !twoWayBindings.some(binding => binding.name === input.name),
+    );
+    const outputs = allOutputs.filter(
```

Comment:
Why are we filtering these out when all the bindings are being concatenated into the same array at the end?

Reply:
Two way bindings will show up in the `inputs` and `outputs` of a node. To make things easier to reason about I dedup them and separated them into their own category to make it more clear that they are two-way bindings

Reply:
Sure, but it seems weird to me that we separate them into 3 different arrays just to combine them again. We could dedupe them as it's being constructed.

Reply:
The `inputs`, `outputs`, and `attributes` are already separate in the ast. We can remove the inputs and outputs that are two-way bindings as we identify & construct the two-way bindings array, but I think it makes the code harder to read. I pushed a commit that does that, so let me know what you think

Reply:
Okay, I see what you mean with your first approach being more readable. Feel free to go with whatever makes sense to you.